src/Chirp.Infrastructure/CheepRepository.cs

```diff
@@ -185,29 +187,29 @@ public int GetCountOfAllCheepFromAuthor(string author)
     public async Task<Cheep> Create(CheepCreateDTO cheep)
     {
         //NullReferenceException is handled in the constructor - CheepRepository()
-        var validationResult = await _validator.ValidateAsync(cheep);
+        ValidationResult result = _validator.Validate(cheep);
 
-        if (!validationResult.IsValid)
+        if (!result.IsValid)
         {
             throw new ValidationException();
         }
+        
+        var user = db.Authors.SingleAsync(u => u.Name == cheep.Author);
 
-        var user = await db.Authors.SingleAsync(u => u.Name == cheep.Author);
-
-        var entity = new Cheep{
-            Text = cheep.Text,
-            Author = user,
-            TimeStamp = DateTime.Now
-        };
-
-        db.Add(new Cheep
+        try
+        {
+             db.Add(new Cheep
                 {
-                    Author = entity.Author,
-                    Text = entity.Text,
-                    TimeStamp = entity.TimeStamp
+                    Author = user.Result,
+                    Text = cheep.Text,
+                    TimeStamp = DateTime.Now
                 });
-        await db.SaveChangesAsync();
-
-        return entity;
+        }
+        catch (System.Exception)
+        {
+            throw;
+        }
+        db.SaveChanges();
+        return null;
```

Comment:
We do we return null, no matter what.
Couldn't we just change the return type to a Task, and not Task<Cheep>?

And do we use the await keyword inside?
If not, does the method have to be async? If not, the return type could simply be void.

Reply:
The async method is what Rasmus told us it had to be, and the return null is what the tests depend on

And no theres no await keyword, but as you can see theres no exceptions demanding an await, which is why its nessesary to use it, the method as it is will not work without, and right now it works as we inteded

Reply:
But this method runs synchronously, since there is no await keyword.
Therefor it is ambigious to declare it async, and the keyword should be removed.

Reply:
And also I still think we should change the metods return type.
It seems odd to return null statically.

Reply:
I have made the described changes in commit 34b4ea4